bridge-relayer/relayer.go

```diff
@@ -0,0 +1,684 @@
+	}
+
+	bridgeConfig := consensusConfig.Bridge[*sopts.externalChainID]
+
+	txRelayer, err = txrelayer.NewTxRelayer(txrelayer.WithIPAddress(bridgeConfig.JSONRPCEndpoint))
+	if err != nil {
+		return nil, errFunc(err)
+	}
+
+	relayer.externalRPCAddr = bridgeConfig.JSONRPCEndpoint
+	relayer.externalClient = txRelayer
+
+	relayer.externalChainID = big.NewInt(int64(*sopts.externalChainID))
+
+	// address of the bridge storage should also be read from the genesis
+	relayer.bridgeStorageAddr = contracts.BridgeStorageContract
+	relayer.internalGatewayAddr = bridgeConfig.InternalGatewayAddr
+	relayer.externalGatewayAddr = bridgeConfig.ExternalGatewayAddr
+
+	relayer.pollInterval = time.Second * 5
+
+	pkBytes, err := hex.DecodeString(privateKey)
+	if err != nil {
+		return nil, errFunc(err)
+	}
+
+	x, y := btcec.S256().ScalarBaseMult(pkBytes)
+
+	pk := &ecdsa.PrivateKey{
+		D: new(big.Int).SetBytes(pkBytes),
+		PublicKey: ecdsa.PublicKey{
+			Curve: btcec.S256(),
+			X:     x,
+			Y:     y,
+		},
+	}
+
+	logger, err := newLoggerFromConfig(sopts)
+	if err != nil {
+		return nil, err
+	}
+
+	relayer.logger = logger
+
+	relayer.privateKey = crypto.NewECDSAKey(pk)
+
+	if sopts.dbPath == nil {
+		relayer.db, err = bolt.Open("bridge-relayer.db", 0600, nil)
+	} else {
+		relayer.db, err = bolt.Open(*sopts.dbPath, 0600, nil)
+	}
+
+	if err != nil {
+		return nil, errFunc(err)
+	}
+
+	err = relayer.db.Update(func(tx *bolt.Tx) error {
+		_, err := tx.CreateBucketIfNotExists([]byte("lastBridgedBucket"))
+		if err != nil {
+			return err
+		}
+
+		return nil
+	})
+
+	if err != nil {
+		return nil, errFunc(err)
+	}
+
+	return relayer, nil
+}
+
+func (r *BridgeRelayer) Start() {
+	var lastBridged = big.NewInt(-1)
+
+	key := []byte{'l', 'a', 's', 't'}
+
+	err := r.db.Update(func(tx *bolt.Tx) error {
+		bucket := tx.Bucket([]byte("lastBridgedBucket"))
+
+		if bucket == nil {
+			return errors.New("cannot find a bucket with the `lastBridgedBucket` name")
+		}
+
+		value := bucket.Get(key)
+
+		if value != nil {
+			var temp string
+			if err := json.Unmarshal(value, &temp); err != nil {
+				return err
+			}
+
+			if _, ok := lastBridged.SetString(temp, 10); !ok {
+				return errors.New("cannot set lastBridge correctly")
+			}
+		}
+
+		return nil
+	})
+
+	if err != nil {
+		fmt.Println(err)
+
+		return
+	}
+
+	t := time.NewTicker(r.pollInterval)
+
+	r.logger.Info("Starting the bridge relayer", "start batch id", lastBridged.String())
+
+	for {
+		select {
+		case <-t.C:
+			r.logger.Info("Trying to get a batches", "the id higher than", lastBridged.String())
+
+			batches, err := GetBridgeBatchesFromNumber(big.NewInt(0).Add(lastBridged, big.NewInt(1)), r.internalClient)
+			if err != nil {
+				r.logger.Error("failed to get batches from BridgeStorage contract", "err", err)
+
+				continue
+			} else {
+				if len(batches) == 0 {
+					r.logger.Info("Cannot find a new batches")
+
+					continue
+				} else {
+					r.logger.Info("Found", len(batches), "new batches")
+				}
```

Comment:
```suggestion
      r.logger.Info("Found", len(batches), "new batches")
```
We don't have to log different logs for this case, found 0 new batches is ok if there are no new batches